js/slideshow.js

```diff
@@ -184,6 +183,95 @@
 		},
 
 		/**
+		 * Determines which colour to use for the background
+		 *
+		 * @param {*} image
+		 * @param {string} mimeType
+		 *
+		 * @returns {string}
+		 * @private
+		 */
+		_getBackgroundColour: function (image, mimeType) {
+			var backgroundColour = this.darkBackgroundColour;
+			if (this._isTransparent(mimeType) && this._isMainlyDark(image)) {
+				backgroundColour = this.lightBackgroundColour;
+			}
+			return backgroundColour;
+		},
+
+		/**
+		 * Calculates the luminance of an image to determine if an image is mainly dark
+		 *
+		 * @param {*} image
+		 *
+		 * @returns {boolean}
+		 * @private
+		 */
+		_isMainlyDark: function (image) {
+			var isMainlyDark = true;
+			var numberOfSamples = 1000; // Seems to be the sweet spot
+			// The name has to be 'canvas'
+			var lumiCanvas = document.createElement('canvas');
+
+			var imgArea = image.width * image.height;
+			var canArea = numberOfSamples;
+			var factor = Math.sqrt(canArea / imgArea);
+
+			var scaledWidth = factor * image.width;
+			var scaledHeight = factor * image.height;
+			lumiCanvas.width = scaledWidth;
+			lumiCanvas.height = scaledHeight;
+			var lumiCtx = lumiCanvas.getContext('2d');
+			lumiCtx.drawImage(image, 0, 0, scaledWidth, scaledHeight);
+			var imgData = lumiCtx.getImageData(0, 0, lumiCanvas.width, lumiCanvas.height);
+			var pix = imgData.data; // pix.length will be approximately 4*numberOfSamples (for RGBA)
+			var pixelArraySize = pix.length;
+			var totalLuminance = 0;
+			var sampleNumber = 1;
+			var averageLuminance;
+			var totalAlpha = 0;
+			var alphaLevel;
+			var red = 0;
+			var green = 0;
+			var blue = 0;
+			var alpha = 0;
+			var lum = 0;
+			var alphaThreshold = 0.1;
+
+			var sampleCounter = 0;
+			var itemsPerPixel = 4; // red, green, blue, alpha
+			// i += 4 because 4 colours for every pixel
+			for (var i = 0, n = pixelArraySize; i < n; i += itemsPerPixel) {
+				sampleCounter++;
+				alpha = pix[i + 3] / 255;
+				totalAlpha += alpha;
+				if (Math.ceil(alpha * 100) / 100 > alphaThreshold) {
+					red = pix[i];
+					green = pix[i + 1];
+					blue = pix[i + 2];
+					// Luminance formula from
+					// http://stackoverflow.com/questions/596216/formula-to-determine-brightness-of-rgb-color
+					lum = (red + red + green + green + green + blue) / 6;
+					//lum = (red * 0.299 + green * 0.587 + blue * 0.114 );
+					totalLuminance += lum * alpha;
+					sampleNumber++;
+				}
+			}
+
+			// Deletes the canvas
+			lumiCanvas = null;
+
+			// Calculate the optimum background colour for this image
+			averageLuminance = Math.ceil((totalLuminance / sampleNumber) * 100) / 100;
+			alphaLevel = Math.ceil((totalAlpha / numberOfSamples) * 100);
+			if (averageLuminance < 50 && alphaLevel < 90) {
+				isMainlyDark = false;
```

Comment:
I think it should be `isMainlyDark = true;` here.

A further refactoring could help to understand:
```js
var hasLowLuminance = (averageLuminance < 50);
var hasTransparency = (alphaLevel < 90);
return hasLowLuminance && hasTransparency;
```